app/controllers/CreditAndRefundController.scala

```diff
@@ -188,32 +188,28 @@ class CreditAndRefundController @Inject()(val authorisedFunctions: FrontendAutho
 
       case financialDetailsModel: List[FinancialDetailsModel] =>
         val balance: Option[BalanceDetails] = financialDetailsModel.headOption.map(balance => balance.balanceDetails)
-        repaymentService.start(user.nino, balance.flatMap(_.availableCredit).getOrElse(BigDecimal(0))).flatMap { repayment =>
-          repayment match {
-            case RepaymentJourneyModel(nextUrl) =>
-              Future.successful(Redirect(nextUrl))
-            case RepaymentJourneyErrorResponse(status, message) =>
-              Logger("application")
-                .error(s"[CreditAndRefundController][handleRefundRequest] - failed RepaymentJourney: $status ")
-              Future.successful(itvcErrorHandler.showInternalServerError())
-          }
+        repaymentService.start(user.nino, balance.flatMap(_.getAbsoluteAvailableCreditAmount).getOrElse(BigDecimal(0))).flatMap {
```

Comment:
there is an outstanding question if we really should pass zero amount, but I am didn't get answer to this yet, so we can leave it as is;